event-store-derive-internals/src/lib.rs

```diff
@@ -12,4 +12,6 @@ pub trait EventData: Serialize + DeserializeOwned {
 }
 
 /// Trait implemented on the events enum
-pub trait Events: Serialize + DeserializeOwned {}
+pub trait Events: Serialize + DeserializeOwned {
+    fn event_type(&self) -> &'static str;
```

Comment:
Can you add `event_namespace` and `event_namespace_and_type` so the same funcs are available on enums as they are structs?

Actually, to go a bit further, why not put them in their own trait (`EventIdentifier` or something) and add that as a bound, so `Events` becomes `Events: Serialize + DeserializeOwned + EventIdentifier`. Thoughts?

Reply:
Sounds good